libcxx/include/fstream

```diff
@@ -841,8 +849,9 @@ typename basic_filebuf<_CharT, _Traits>::int_type basic_filebuf<_CharT, _Traits>
 
   if (__always_noconv_) {
     size_t __n = static_cast<size_t>(this->pptr() - this->pbase());
-    if (std::fwrite(this->pbase(), sizeof(char_type), __n, __file_) != __n)
-      return traits_type::eof();
+    if (std::fwrite(this->pbase(), sizeof(char_type), __n, __file_) != __n) {
```

Comment:
IIUC you're currently only testing this path. We should also test the `fwrite` calls in the `else` path (i.e. have a locale that's not `__always_noconv_` and have that return the different possible states).